### PR TITLE
Replace AssertJ's deprecated asList() DSL method

### LINKED ISSUE #3270: CronJobHandlerTest : Replace AssertJ's deprecated `asList()` DSL method with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`
### Component

JKube Kit

### Description

In the latest versions of Assertj `asList()` DSL method has been marked as deprecated:

https://github.com/eclipse-jkube/jkube/blob/d000d3ffdc1a18047df64f772391d8829f5b3cda/jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/handler/CronJobHandlerTest.java#L100

users are advised to use `asInstanceOf(InstanceOfAssertFactories.list(type.class))` instead, we need to update `asList()` calls with this. 

### Expected Behavior
AssertJ's `asList()` is replaced with `asInstanceOf(InstanceOfAssertFactories.list(type.class))`

### Acceptance Criteria
- [ ] `asList()` DSL method call is replaced with  `asInstanceOf(InstanceOfAssertFactories.list(type.class))` in all places

### PR DESCRIPTION
**Pull request by ivamly**
CronJobHandlerTest : Replace AssertJ's deprecated asList() DSL method with asInstanceOf(InstanceOfAssertFactories.list(type.class))

## Description

Fixes #3270

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3271 (_2024-07-25T06:49:13Z_)

:gear: [JKube E2E Tests (10089501176)](https://github.com/eclipse-jkube/ci/actions/runs/10089501176) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3271)](https://github.com/eclipse-jkube/ci/actions/runs/10089501176/job/27897068517)

File: jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/handler/CronJobHandlerTest.java [hunk 1]
```diff
@@ -97,7 +98,7 @@ void get_withValidConfigs_shouldReturnConfigWithContainers() {
             .extracting(JobSpec::getTemplate)
             .extracting(PodTemplateSpec::getSpec)
             .hasFieldOrPropertyWithValue("restartPolicy", "Never")
-            .extracting(PodSpec::getVolumes).asList().first()
+            .extracting(PodSpec::getVolumes).asInstanceOf(InstanceOfAssertFactories.list(List.class)).first()
```
Comment (rohanKanojia) [inline jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/handler/CronJobHandlerTest.java:101]: type inside `list(...)` should be type of List enclosed in `extracting`.

In this case it should be `Volume`

Comment (ivamly) [reply jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/handler/CronJobHandlerTest.java]: Thank you for help! I fixed it.

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**